<launch>

  <arg name="stdout" default="screen" />
  <arg name="camera_name" default="/zed2i/zed_node/" />
  <arg name="image_topic" default="image_rect_color" />
  <arg name="frame_name" default="zed2i_left_camera_optical_frame"/>
  <arg name="sensor_id" default="0" />

  <!-- Circle detection left -->
  <node name="circle_detection_left" pkg="opencv_apps" type="hough_circles" output="$(arg stdout)" >
    <remap from="image" to="$(arg camera_name)/left/$(arg image_topic)" />
    <remap from="~circles" to="circle_detection_left/left_circles" />
    <rosparam>
      use_camera_info: True
      debug_view: fasle
      canny_threshold: 96
      accumulator_threshold: 45
      gaussian_blur_size: 9
      gaussian_sigma_x: 1
      gaussian_sigma_y: 1
      dp: 2
      min_circle_radius: 28
      max_circle_radius: 32
      queue_size: 4
    </rosparam>
  </node>

  <!-- Circle detection right -->
  <node name="circle_detection_right" pkg="opencv_apps" type="hough_circles" output="$(arg stdout)" >
    <remap from="image" to="$(arg camera_name)/right/$(arg image_topic)" />
    <remap from="~circles" to="circle_detection_right/right_circles" />
    <rosparam>
      use_camera_info: True
      debug_view: fasle
      canny_threshold: 96
      accumulator_threshold: 45
      gaussian_blur_size: 9
      gaussian_sigma_x: 1
      gaussian_sigma_y: 1
      dp: 2
      min_circle_radius: 28
      max_circle_radius: 32
      queue_size: 4
    </rosparam>
  </node>

  <!-- Our node -->
  <node pkg="velo2cam_stereoHough" type="stereo_pattern" name="stereo_pattern_$(arg sensor_id)" output="screen"></node>

  <!-- Per ruotare la camera da capire che farne -->
  <node pkg="tf" type="static_transform_publisher" name="stereo_ros_tf" args="0 0 0 -1.57079632679 0 -1.57079632679 rotated_$(arg frame_name) $(arg frame_name) 100"/>
</launch>
